File: TestProject-java/app/src/main/java/com/azman/testproject_java/SecondActivity.java
package com.azman.testproject_java;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        // enable back button on second page
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null){
            actionBar.setTitle("Second Activity X");
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        Button button2 = findViewById(R.id.button2);
        button2.setOnClickListener(view -> {
            Intent intent = new Intent(SecondActivity.this, MainActivity.class );
            startActivity(intent);
        });
    }
}

File: TestProject-kt/app/src/main/java/com/azman/testproject_kt/SecondActivity.kt
package com.azman.testproject_kt

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        // enable back button on second page
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.title = "Second Activity"
            actionBar.setDisplayHomeAsUpEnabled(true)
        }

        val secondActivityBtn: Button = findViewById(R.id.button2nd)
        secondActivityBtn.setOnClickListener {
            val intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
        }
    }
}